src/main/java/com/varchenko/cache/LRUCache.java

```diff
@@ -0,0 +1,45 @@
+package com.varchenko.cache;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class LRUCache<K, V> implements Cache<K, V> {
+
+    private static final int DEFAULT_CAPACITY = 5;
+    private final Map<K, V> cacheMap;
+    private int capacity;
+
+    public LRUCache(int capacity) {
+        this();
+        this.capacity = capacity;
+    }
+
+    public LRUCache() {
+        this.capacity = DEFAULT_CAPACITY;
+        this.cacheMap = new LinkedHashMap<>(capacity);
+    }
+
+    @Override
+    public void put(K key, V value) {
+        if (size() > capacity) {
+            removeEldest();
+            cacheMap.put(key, value);
+        }
+        cacheMap.put(key, value);
+    }
+
+    @Override
+    public V get(K key) {
+        return cacheMap.get(key);
+    }
+
+    @Override
+    public int size() {
+        return cacheMap.size();
+    }
+
+    private void removeEldest() {
+        Map.Entry<K, V> eldest = cacheMap.entrySet().iterator().next();
```

Comment:
Do you just remove the first element in the cacheMap?

Reply:
done